src/components/App.tsx

```diff
@@ -776,10 +788,36 @@ class App extends React.Component<AppProps, AppState> {
 
     if ("ResizeObserver" in window && this.excalidrawContainerRef?.current) {
       this.resizeObserver = new ResizeObserver(() => {
+        // compute isMobile state
+        // ---------------------------------------------------------------------
+        const {
+          width,
+          height,
+        } = this.excalidrawContainerRef.current!.getBoundingClientRect();
+        this.isMobile =
+          width < MQ_MAX_WIDTH_PORTRAIT ||
+          (height < MQ_MAX_HEIGHT_LANDSCAPE && width < MQ_MAX_WIDTH_LANDSCAPE);
+        // refresh offsets
+        // ---------------------------------------------------------------------
         this.updateDOMRect();
       });
       this.resizeObserver?.observe(this.excalidrawContainerRef.current);
+    } else {
+      const isMobileQuery = `(max-width: ${MQ_MAX_WIDTH_PORTRAIT}px), (max-height: ${MQ_MAX_HEIGHT_LANDSCAPE}px) and (max-width: ${MQ_MAX_WIDTH_LANDSCAPE}px)`;
+
+      const matchMedia = window.matchMedia
+        ? window.matchMedia(isMobileQuery)
+        : (({
+            matches: false,
+            addListener: () => {},
+            removeListener: () => {},
+          } as any) as MediaQueryList);
+
+      const handler = () => (this.isMobile = matchMedia.matches);
+      matchMedia.addListener(handler);
+      this.detachIsMobileMqHandler = () => matchMedia.removeListener(handler);
```

Comment:
```suggestion
      if (window.matchMedia) {
        const mediaQuery = window.matchMedia(isMobileQuery);
        const handler = () => (this.isMobile = mediaQuery.matches);
        mediaQuery.addListener(handler);
        this.detachIsMobileMqHandler = () => mediaQuery.removeListener(handler);
      }
```

Reply:
Good point. I simply ported the code from the compo from before.